### PR TITLE
Changes for Round 17

### PR DESCRIPTION
**Pull request by internetfett**
- Added a per-round Global Forum for all dominions (basically a carbon-copy of Council with a few additional features to help with self-moderation)
- Restart feature for dominions in hour 0 of protection (no race change yet)
- Additional resources for late signups
- GPC Balance Changes (see changelog or #gpc-announcements)
- User Valhalla page linked from other categories
- Lots of information updates on various pages
- Additional statistics

### CODE REVIEW

**Comment by internetfett** [comment]
Screen Shot 2020-02-21 at 5 04 43 PM
Screen Shot 2020-02-26 at 10 55 59 PM
Screen Shot 2020-02-25 at 2 15 57 PM

File: app/resources/views/pages/dominion/advisors/magic.blade.php [hunk 1]
```diff
@@ -38,7 +38,7 @@
                                     <td>{{ $spellInfo['description'] }}</td>
                                     <td class="text-center">{{ $spell->duration }}</td>
                                     <td class="text-center">
-                                        @if ($spell->cast_by_dominion_id !== null)
+                                        @if ($spell->cast_by_dominion_id !== null && ($spell->cast_by_dominion_id == $selectedDominion->id || $spellCalculator->isSpellActive($selectedDominion, 'surreal_perception')))
```
Comment (WaveHack) [inline app/resources/views/pages/dominion/advisors/magic.blade.php]: If you don't have Surreal Perception active, someone casts a spell on you, and you cast SP afterwards, you're able to see in retrospect who cast that spell [on you].

Is this intended behavior? Or something to fix later? We can leave it like this for now regardless

Comment (internetfett) [reply app/resources/views/pages/dominion/advisors/magic.blade.php]: Yes, that's how I intended it. I think it makes sense thematically and makes the spell more useful (which is the primary goal of the GPC's changes to it).

Currently, you can always see who cast it (and more importantly, your enemies can too), which is not how it was in DC. GPC can discuss this again after a round to try it out. I think that will be standard going forward, to review all changes from the previous round and see how they worked out.

Comment (WaveHack) [reply app/resources/views/pages/dominion/advisors/magic.blade.php]: Alright no worries. Just checking if it's not an oversight. If the GPC likes it we keep it

File: app/resources/views/pages/valhalla/round-type.blade.php [hunk 2]
```diff
@@ -35,7 +35,11 @@
                         <tr>
                             @foreach ($row as $column => $value)
                                 <td{!! (isset($headers[$column]['align-center']) && $headers[$column]['align-center']) ? ' class="text-center"' : null !!}>
-                                    {{ $value }}
+                                    @if ($column == 'player')
+                                        {!! $value !!}
```
Comment (WaveHack) [inline app/resources/views/pages/valhalla/round-type.blade.php]: Why was this added?

Comment (internetfett) [reply app/resources/views/pages/valhalla/round-type.blade.php]: I added a link to the 'lifetime' page which shows land/networth for all of that users' rounds played. Since the template is the same for all of the Valhalla pages and the displayed data has to be generated in a map function in the Controller, the link is generated added there and has to be escaped to appear on the page. I only did it for the player column just to prevent any unintended side-effects.

Comment (WaveHack) [reply app/resources/views/pages/valhalla/round-type.blade.php]: I'm slightly concerned some players try be funny and put HTML in their dom name, which would normally be 'fine' (since it gets escaped everywhere), but then except here. Would this be a risk? I think you have a better grasp of the context regarding this than I do atm 😋

Comment (internetfett) [reply app/resources/views/pages/valhalla/round-type.blade.php]: Looks like I tested that. It escapes (somewhat) the player name within the link that gets generated. Not sure if that's 100% safe, but it worked for simple html/javascript stuff I tried.

_htmlentities($dominion->user->display_name)_
